SharedSequence/src/test/kotlin/org/notests/sharedsequence/DriverTest.kt

```diff
@@ -0,0 +1,835 @@
+
+    assertEquals(listOf(next(0, 1),
+                        next(0, 2),
+                        next(0, -1),
+                        complete(0)),
+                 observer.recordedEvents())
+
+    val expectedEvents = arrayOf(1, 2, -1)
+    assertArrayEquals(expectedEvents, events)
+  }
+
+  @Test
+  fun testAsDriver_distinctUntilChanged1() {
+    val observable = scheduler.createColdObservable<Int>(next(0, 1),
+                                                         next(0, 2),
+                                                         next(0, 2),
+                                                         error(0, Error("Test")))
+    observable.asDriver(-1)
+      .distinctUntilChanged()
+      .drive(observer)
+
+    scheduler.advanceTimeBy(1)
+
+    assertEquals(listOf(next(0, 1),
+                        next(0, 2),
+                        next(0, -1),
+                        complete(0)),
+                 observer.recordedEvents())
+  }
+
+  @Test
+  fun testAsDriver_distinctUntilChanged2() {
+    val observable = scheduler.createColdObservable<Int>(next(0, 1),
+                                                         next(0, 2),
+                                                         next(0, 2),
+                                                         error(0, Error("Test")))
+    observable.asDriver(-1)
+      .distinctUntilChanged { e -> e }
+      .drive(observer)
+
+    scheduler.advanceTimeBy(1)
+
+    assertEquals(listOf(next(0, 1),
+                        next(0, 2),
+                        next(0, -1),
+                        complete(0)),
+                 observer.recordedEvents())
+  }
+
+  @Test
+  fun testAsDriver_distinctUntilChanged3() {
+    val observable = scheduler.createColdObservable<Int>(next(0, 1),
+                                                         next(0, 2),
+                                                         next(0, 2),
+                                                         error(0, Error("Test")))
+    observable.asDriver(-1)
+      .distinctUntilChanged { e1, e2 -> e1 == e2 }
+      .drive(observer)
+
+    scheduler.advanceTimeBy(1)
+
+    assertEquals(listOf(next(0, 1),
+                        next(0, 2),
+                        next(0, -1),
+                        complete(0)),
+                 observer.recordedEvents())
+  }
+
+  @Test
+  fun testAsDriver_flatMap() {
+    val observable = scheduler.createColdObservable<Int>(next(0, 1),
+                                                         next(0, 2),
+                                                         error(0, Error("Test")))
+    observable.asDriver(-1)
+      .flatMapDriver {
+        Driver.just(it + 1)
+      }
+      .drive(observer)
+
+    scheduler.advanceTimeBy(1)
+
+    assertEquals(listOf(next(0, 2),
+                        next(0, 3),
+                        next(0, 0),
+                        complete(0)),
+                 observer.recordedEvents())
+  }
+
+  @Test
+  fun testAsDriver_mergeSync() {
```

Comment:
What is this thing doing?

Reply:
aha before I had more `merge` methods. Will remove this test